### PR TITLE
[WiP] Windows port

### PR DESCRIPTION
**Pull request by SeanCampbell**
Merges kele's [Win32 port](https://github.com/Stackdriver/collectd/commit/09f43256e6e95df8108a7a1791b4ff81690b1960) into upstream collectd.

### CODE REVIEW

File: src/daemon/collectd.c [hunk 1]
```diff
@@ -680,7 +712,6 @@ int main(int argc, char **argv) {
       exit_status = 1;
     }
   } else {
-    INFO("Initialization complete, entering read-loop.");
```
Comment (rpv-tomsk) [inline src/daemon/collectd.c]: Why did you change this?

File: src/daemon/common.c [hunk 2]
```diff
@@ -262,10 +266,7 @@ int swrite(int fd, const void *buf, size_t count) {
   pfd.events = POLLIN | POLLHUP;
   pfd.revents = 0;
   if (poll(&pfd, 1, 0) > 0) {
-    char buffer[32];
-    if (recv(fd, buffer, sizeof(buffer), MSG_PEEK | MSG_DONTWAIT) == 0) {
-      /* if recv returns zero (even though poll() said there is data to be
-       * read), that means the connection has been closed */
+    if (pfd.revents & POLLHUP) {
```
Comment (rpv-tomsk) [inline src/daemon/common.c]: As stated before, this will not work.

File: src/utils_latency.c [hunk 3]
```diff
@@ -156,7 +156,9 @@ void latency_counter_add(latency_counter_t *lc, cdtime_t latency) /* {{{ */
     change_bin_width(lc, latency);
     bin = (latency - 1) / lc->bin_width;
     if (bin >= HISTOGRAM_NUM_BINS) {
-      ERROR("utils_latency: latency_counter_add: Invalid bin: %" PRIu64, bin);
+      ERROR("utils_latency: latency_counter_add: Invalid bin: %"
```
Comment (rpv-tomsk) [inline src/utils_latency.c]: This looks strange for me...

**Comment by SeanCampbell** [comment]
Thanks, Pavel. This branch is very much a work in progress, and I'm splitting it into multiple PRs focused on specific changes for review.

I have it as a PR so that I can tell if it's passing the build checks on the collectd repo, and octo suggested [WiP] in the name of the PR would indicate it isn't yet ready for review.
